### DISCUSSION TITLE
Is there a log file for a "Crash Report"

### DISCUSSION THREAD

**Discussion by pedro-w** [lxqt/lxqt#2315] [General]
I've just today started seeing a dialog box when I log on

    Crash Report
    Desktop has crashed too many times. Its autorestart has been disabled until next login.

In itself this is no use to me either trying to fix it or reporting a possible bug. Is there a log file somewhere which contains more detail? I tried `journalctl` and `systemctl status` but I don't know, assuming it is something managed by systemd, what the  service name is.

(This is Fedora 36 running LXQT on VirtualBox 7)

**Comment by pedro-w** [comment]
To reply to myself, I realised Desktop is an LXQt module, `/etc/xdg/autostart/lxqt-desktop.desktop` not a systemd thing. But I still don't know where the crash log file might be.

**Reply by stefonarch** [reply]
What do you get if you start it in terminal?
`pcmanfm-qt -d`

**Reply by pedro-w** [reply]
I get 
```
$ pcmanfm-qt -d
Segmentation fault (core dumped)
```

**Comment by tsujan** [comment]
> Is there a log file somewhere which contains more detail?

Crashes create coredumps, which can be used to provide backtraces. Consult your distro doc.

**Reply by pedro-w** [reply]
OK, great, that's very useful.


For reference I'll include the BT here but I am pretty sure it's connected to VBox guest additions so I'll pester them first (unless you want me to file a bug here too?)
```
3365        return R->name_len;                                                                                             
[Current thread is 1 (Thread 0x7ff93fe38580 (LWP 3979))]
(gdb) bt
#0  xcb_get_atom_name_name_length (R=0x0) at /usr/src/debug/libxcb-1.13.1-9.fc36.x86_64/src/xproto.c:3365
#1  0x00007ff9419ac47e in XdndWorkaround::atomName (atom=0)
    at /usr/src/debug/libfm-qt-1.0.0-5.fc36.x86_64/src/xdndworkaround.cpp:110
#2  0x00007ff9419b1ff0 in XdndWorkaround::clientMessage (event=0x556e98fc1500, this=0x556e98fe8170)
    at /usr/src/debug/libfm-qt-1.0.0-5.fc36.x86_64/src/xdndworkaround.cpp:165
#3  XdndWorkaround::nativeEventFilter (this=0x556e98fe8170, eventType=..., message=0x556e98fc1500)
    at /usr/src/debug/libfm-qt-1.0.0-5.fc36.x86_64/src/xdndworkaround.cpp:85
#4  0x00007ff9404a5344 in QAbstractEventDispatcher::filterNativeEvent (this=<optimized out>, eventType=..., 
    message=message@entry=0x556e98fc1500, result=result@entry=0x7fff9e777bf8) at kernel/qabstracteventdispatcher.cpp:495
#5  0x00007ff92f8efba3 in QXcbConnection::handleXcbEvent (this=this@entry=0x556e98f219b0, event=event@entry=0x556e98fc1500)
    at qxcbconnection.cpp:579
#6  0x00007ff92f8f1276 in QXcbConnection::processXcbEvents (this=0x556e98f219b0, flags=...) at qxcbconnection.cpp:1065
#7  0x00007ff92f9178b7 in xcbSourceDispatch (source=<optimized out>) at qxcbeventdispatcher.cpp:103
#8  0x00007ff9417bbfaf in g_main_dispatch (context=0x7ff928005010) at ../glib/gmain.c:3417
#9  g_main_context_dispatch (context=0x7ff928005010) at ../glib/gmain.c:4135
#10 0x00007ff9418112c8 in g_main_context_iterate.constprop.0 (context=context@entry=0x7ff928005010, block=block@entry=1, 
    dispatch=dispatch@entry=1, self=<optimized out>) at ../glib/gmain.c:4211
#11 0x00007ff9417b9940 in g_main_context_iteration (context=0x7ff928005010, may_block=1) at ../glib/gmain.c:4276
#12 0x00007ff9404f902a in QEventDispatcherGlib::processEvents (this=0x556e98fda9d0, flags=...)
    at kernel/qeventdispatcher_glib.cpp:423
#13 0x00007ff9404a6c1a in QEventLoop::exec (this=this@entry=0x7fff9e777ee0, flags=..., flags@entry=...)
```

**Reply by tsujan** [reply]
Thanks! The backtrace shows a crash in libfm-qt. A nullity check (actually, two) is missing in `xdndworkaround.cpp` — I'll add it soon.

But I have no idea why `xcb_get_atom_name_reply` should give `NULL` in your case. It had never happened before.

**Reply by tsujan** [reply]
> (unless you want me to file a bug here too?)

Yes, please, before I make a PR.

**Comment by pedro-w** [comment]
@tsujan 
Will do, should it be in [this repo](https://github.com/lxqt/lxqt/issues) or in the [pcmanfm-qt](https://github.com/lxqt/pcmanfm-qt/issues) one?

**Reply by tsujan** [reply]
> ... should it be in [this repo](https://github.com/lxqt/lxqt/issues) or in the [pcmanfm-qt](https://github.com/lxqt/pcmanfm-qt/issues) one?

In https://github.com/lxqt/pcmanfm-qt/issues, please.

**Comment by pedro-w** [comment]
Done, https://github.com/lxqt/pcmanfm-qt/issues/1668

Thanks for your help!

**Reply by tsujan** [reply]
Thanks for the report!
